bookdown/08-interpretation.Rmd

```diff
@@ -1,10 +1,165 @@
+
+* `Shapley` computes feature contributions for single predictions with the Shapley value -- an approach from cooperative game theory ([Shapley Value](https://christophm.github.io/interpretable-ml-book/shapley.html)).
+
+
+**FeatureEffects**
+
+In addition to the commands above the following two need to be ran:
+
+```{r eff, message=FALSE, warning=FALSE, fig.cap='Plot of the results from FeatureEffects. FeatureEffects computes and plots feature effects of prediction models',  fig.align='center'}
+effect = FeatureEffects$new(model)
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+`effect` defines the interpretation model by wrapping `Predictor`. After this is done the selected features can be plotted. In this example, all of the features provided by the `palmerpenguins` data set were used. 
+
+
+All features outside of `year` provide meaningful interpretable information. It should be clear why `year` doesn't provide anything of significance. `bill_length_mm` shows for example that when the bill length is smaller than roughly 40mm, there is a high chance that the penguin is an Adelie. 
+
+**Shapley**
+  
+```{r mod, message=FALSE, warning=FALSE, fig.cap='Plot of the results from Shapley. $\\phi$ gives the increase or decrease in probabilty given the values on the vertical axis',  fig.align='center'}
+x = penguins[which(names(penguins) != "species")]
+model = Predictor$new(learner, data = penguins, y = "species")
+x.interest <-data.frame(penguins[1,])
+shapley <- Shapley$new(model, x.interest = x.interest)
+plot(shapley)
+```
+
+The $\phi$ provides insight into the probability given the values on the vertical axis. For example, a penguin is less likely to be Gentoo if the bill_depth_=18.7 is and much more likely to be Adelie than Chinstrap.
+
+
+**FeatureImp**
+
+```{r imp, message=FALSE, warning=FALSE, fig.cap='Plot of the results from FeatureImp. FeatureImp visualizes the importance of features given the prediction model',  fig.align='center'}
+effect = FeatureImp$new(model, loss = "ce")
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+
+`FeatureImp` shows the level of importance of the features when classifying the penguins. It is clear to see that the `bill_length_mm` is of high importance and one should concentrate on different boundaries of this feature when attempting to classify the three species. 
+
+
+
+**Predicting**
+  
+It is also interesting to see how well a model has trained a data set. For this section, exactly as was recommended in section [2.4](https://mlr3book.mlr-org.com/train-predict.html) of the mlr3book, 80% of the penguin data set will be used for the training set and 20% for the test set:
```

Comment:
how well a model has trained a data set -> how well the model performs on a data set